### PR TITLE
Add availabilityCheckerInterface to check sufficient variants in stock

### PR DESCRIPTION
**Pull request by Wojdylak**

### CODE REVIEW

File: src/Command/Wishlist/AddProductsToCartInterface.php [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<?php
+
```
Comment (mpysiak) [inline src/Command/Wishlist/AddProductsToCartInterface.php:2]:
```suggestion
/*
 * This file has been created by developers from BitBag.
 * Feel free to contact us once you face any issues or want to start
 * You can find more information about us on https://bitbag.io and write us
 * an email on hello@bitbag.io.
 */
```

**Review by mpysiak — CHANGES_REQUESTED** [review]

**Review by mpysiak — APPROVED** [review]

**Review by lruozzi9 — COMMENTED** [review]
This PR contains various BC on class AddProductsToCartHandler like adding a not nullable argument in the constructor and changing the argument interface of some methods...

**Comment by Wojdylak** [comment]
@lruozzi9 Yes, you are right. I fixed the constructor, but I left the interfaces in private methods.

File: src/CommandHandler/Wishlist/AddProductsToCartHandler.php [hunk 2]
```diff
@@ -59,23 +64,27 @@ private function addProductsToWishlist(Collection $wishlistProducts): void
...
 
-        return $this->productIsInStock($cartItem) && $this->productHasPositiveQuantity($cartItem);
+        return $this->productIsStockSufficient($cartItem) && $this->productHasPositiveQuantity($cartItem);
     }
 
-    private function productIsInStock(OrderItemInterface $product): bool
+    private function productIsStockSufficient(OrderItemInterface $product): bool
     {
-        if ($product->getVariant()->isInStock()) {
+        if (null !== $this->availabilityChecker) {
```
Comment (mpysiak) [inline src/CommandHandler/Wishlist/AddProductsToCartHandler.php:76]: you can merge these two if statements

File: src/CommandHandler/Wishlist/AddProductsToCartHandler.php [hunk 3]
```diff
@@ -59,23 +64,27 @@ private function addProductsToWishlist(Collection $wishlistProducts): void
...
+            if ($this->availabilityChecker->isStockSufficient($product->getVariant(), $product->getQuantity())) {
+                return true;
+            }
+        } elseif ($product->getVariant()->isInStock()) {
```
Comment (mpysiak) [inline src/CommandHandler/Wishlist/AddProductsToCartHandler.php:80]: is elseif needed?

**Review by mpysiak — APPROVED** [review]

**Pull request merged**